.github/actions/write-credentials/action.yml

```diff
@@ -0,0 +1,61 @@
+name: 'write-credentials'
+description: |
+  Collect the needed credentials for testing and place them in the
+  appropriate config directory. Note that all of these secrets are
+  assumed to be base64-encoded JSON service account keys.
+inputs:
+  user-delegated-sa-b64:
+    description: 'Base64-encoded user delegated service account'
+    required: true
+  buffer-client-sa-b64:
+    description: 'Base64-encoded Buffer client service account'
+    required: true
+  testrunner-sa-b64:
+    description: 'Base64-encoded Testrunner service account'
+    required: true
+  testrunner-k8s-sa-b64:
+    description: 'Base64-encoded Testrunner k8s service account'
+    required: true
+  wsm-sa-b64:
+    description: 'Base64-encoded WSM service account'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Write credentials
+      id: 'setup-user-delegated-creds'
+      run: |
+        mkdir -p config
+        USER_DELEGATED_SA_B64=${{ inputs.user-delegated-sa-b64 }}
+        echo ::add-mask::$USER_DELEGATED_SA_B64
```

Comment:
Probably being paranoid here but I want to make sure these secrets (either base64 or decoded). are never erroneously logged either here or in downstream actions.